scripts/security-check.sh

```
@@ -0,0 +1,233 @@
+fi
+
+# 4. 環境変数の使用パターンチェック
+echo "📋 4. Environment variable usage check..."
+while IFS= read -r file; do
+        # process.env の不適切な使用をチェック
+        if grep -nP 'process\.env\.(?!NEXT_PUBLIC_)(?!NODE_ENV\b)[A-Z][A-Z0-9_]+' "$file" >/dev/null; then
+            warning "Non-public environment variable used: $file"
+        fi
+    done < <(find . -path ./node_modules -prune -o -path ./.git -prune -o \( -name "*.ts" -o -name "*.tsx" \) -print)
+    success "Environment variable usage check completed"
+fi
```

Shellcheck (0.10.0)

[error] Parsing stopped here. Is this keyword correctly matched up?

(SC1089)